tests/integration/upgrade/customgateway/update_test.go

```diff
@@ -0,0 +1,409 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gatewayupgrade
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/cluster"
+	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/image"
+	"istio.io/istio/pkg/test/framework/resource"
+	kubetest "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/shell"
+	"istio.io/istio/pkg/test/util/retry"
+	helmtest "istio.io/istio/tests/integration/helm"
+	"istio.io/istio/tests/util"
+)
+
+const (
+	fromVersion          = "1.8.2"
+	aSvc                 = "a" // Used by the default gateway
+	bSvc                 = "b" // Used by the custom ingress gateway
+	IstioNamespace       = "istio-system"
+	retryDelay           = 2 * time.Second
+	RetryTimeOut         = 5 * time.Minute
+	customServiceGateway = "custom-ingressgateway"
+	vsTemplate           = `
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: %s
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - %s
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: %s
+`
+	gwTemplate = `
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: %s
+spec:
+  selector:
+    istio: %s
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+`
+)
+
+type echoDeployments struct {
+	appANamespace, appBNamespace namespace.Instance
+	A, B                         echo.Instances
+}
+
+var (
+	apps              = &echoDeployments{}
+	customGWNamespace namespace.Instance
+	// ManifestPath is path of local manifests which istioctl operator init refers to.
+	ManifestPath = filepath.Join(env.IstioSrc, "manifests")
+)
+
+// TestUpdateWithCustomGateway tests access to an application using an updated
+// control plane and custom gateway
+func TestUpdateWithCustomGateway(t *testing.T) {
```

Comment:
Why do we add a new test instead of extending the existing one?

Reply:
There is no existing tests where we deploy a prior version of the gateway and upgrade it.